lib/elixir/lib/task.ex

```diff
@@ -338,13 +341,17 @@ defmodule Task do
   end
 
   @doc """
-  Receives a group of tasks and a message and finds
+  Takes a list of tasks and a message and finds
   a task that matches the given message.
 
-  This function returns a tuple with the returned value
-  in case the message matches a task that exited with
-  success alongside the matching task. It returns `nil`
-  if no task was found. It exits if the task has failed.
+  If the message matches a task that finished normally,
+  this function returns a tuple with the task's
+  result and `Task` struct.
+
+  If the message matches a task that terminated before
+  sending its result back, this function will exit.
```

Comment:
I still find this behaviour somewhat unsettling. 

When starting a task outside of an OTP behaviour, I can use `Task.Supervisor.async_nolink` and then call `Task.yield` and get back a nice `{:exit, reason}` for a failed task.

However, when spawning tasks inside a gen server, even using `async_nolink`, the user seems to have no choice but to exit when they call `find`.